lib/widgets/swipable_message_row.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+import '../api/model/model.dart';
+import 'icons.dart';
+import 'text.dart';
+import 'theme.dart';
+
+class SwipableMessageRow extends StatefulWidget {
+  const SwipableMessageRow({
+    super.key,
+    required this.children,
+    required this.message,
+  });
+
+  final List<Widget> children;
+  final Message message;
+
+  @override
+  State<StatefulWidget> createState() => _SwipableMessageRowState();
+}
+
+class _SwipableMessageRowState extends State<SwipableMessageRow> {
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Seems like this can just as well be a stateless widget:
```suggestion
  @override
  Widget build(BuildContext context) {
```

(We'll need state here for animation, in #765, but it's easy to add then. And the state might go on a child, rather than this widget itself — one way or another I'd like to arrange it so we don't need a state in the case of MessageEditState.none, at least once the state starts holding other resources like an AnimationController.)